integration/network/ipvlan/ipvlan_test.go

```diff
@@ -425,7 +429,26 @@ func testIpvlanAddressing(client dclient.APIClient) func(*testing.T) {
 	}
 }
 
-// ensure Kernel version is >= v4.2 for ipvlan support
+var (
+	once            sync.Once
+	ipvlanSupported bool
+)
+
+// figure out if ipvlan is supported by the kernel
 func ipvlanKernelSupport() bool {
-	return n.CheckKernelMajorVersionGreaterOrEqualThen(4, 2)
+	if n.CheckKernelMajorVersionGreaterOrEqualThen(4, 2) == false {
+		return false
+	}
+	once.Do(func() {
+		// this may have the side effect of enabling the ipvlan module
+		exec.Command("modprobe", "ipvlan").Run()
+		_, err := os.Stat("/sys/module/ipvlan")
+		if err == nil {
+			ipvlanSupported = true
+		} else if !os.IsNotExist(err) {
+			fmt.Printf("WARNING: ipvlanKernelSupport: stat failed: %v\n", err)
```

Comment:
should we pass `testing.T` to this function, and `t.Logf()` here?

Reply:
Yes, or return an error?

Reply:
It's used as a boolean to skip a test, so I'm not sure that'd help.

If we want to _fail_ in this situation, we could do `t.Errorf()`

Possibly this function should be marked as a `t.Helper()`, but I'm ok with leaving that for now.

Reply:
I meant return an error as well as the boolean. This function is fairly generic (albeit it is not in a generic package) so it seems a little forced to pass it `testing.T`

Reply:
I'm fine either way; it's just that it's implemented as a helper for tests (so messages should ideally be printed through the testing code), and returning an error would mean we'd have to use an intermediate `bool` in the test itself, and handle potential errors there